table/block_based/block_based_table_iterator.cc

```diff
@@ -81,7 +88,7 @@ void BlockBasedTableIterator::SeekImpl(const Slice* target,
     }
   }
 
-  if (need_seek_index) {
+  if (is_first_pass && need_seek_index) {
```

Comment:
After having in-depth understanding of this code path, I realized we don't need to reseek index for async_io during second pass when we just poll the data.